Support suites that have the same option but different extraOptions

armenzg opened the pull request:
Fetching data for dromaeo_dom returns more than one pgo job since it
used to run in the past with different extraOptions (it now runs with e10s & stylo).

armenzg commented:
Thank you very much for volunteering!

My last test is giving me some trouble. I have two objects and I'm getting this as a discrepancy:
```diff
      -          "datetime": [Date: 2018-06-19T10:02:37.000Z]
      +          "datetime": "2018-06-19T10:02:37.000Z"
```
How would you fix that?

I also need to make a minorchange that will guarantee that URLs will have the same sorting for signatureIds.

wlach commented on src/utils/perfherder.js:
```diff
@@ -92,7 +131,18 @@ const subbenchmarksData = async ({ suite, platform }) => {
+  const parentHash = await parentSignatureHash(platform, suite, option);
+  const subtests = await querySubtestsAssociatedToParent(parentHash);
+  const data = await prepareData(subtests);
+  return data;
```
I would personally just `return await prepareData(subtests);` here.

armenzg replied:
I probably did that because I needed a breakpoint to analyze the data. I will fix it.

wlach commented on src/utils/perfherder.js:
```diff
@@ -1,86 +1,125 @@
-  `${TREEHERDER}/api/project/${project}/performance/signatures`
+export const dataPointsEndpointUrl = (project = PROJECT) => (
+  `${TREEHERDER}/api/project/${project}/performance/data/`
 );
```
Someday I would like to add some API endpoints for getting performance data / signatures from treeherder which completely do away with signature hashes, since they're just confusing to people.

armenzg replied:
Is there a bug filed for this? Perhaps it could be a good bug for contributors.

wlach replied:
There is not, it's rather more complex than it sounds so I don't think it would be good for a new comer.

wlach commented:
@armenzg I have no idea where the `datetime` in your expected json is coming from, since I don't see it in the original API output:

https://treeherder.mozilla.org/api/project/mozilla-inbound/performance/data/?framework=1&interval=1209600&signature_id=1640626

armenzg commented:
The data structure is the aftermath of grabbing data from multiple PH endpoints.

The `datetime` property is added [here](https://github.com/mozilla-frontend-infra/perf-goggles-demo/blob/master/src/utils/perfherder.js#L90) to the data fetched from the '[data](https://treeherder.mozilla.org/api/project/mozilla-central/performance/data/?framework=1&interval=7776000&signature_id=1659462)' endpoint.

It is added to be used as the '[x_accessor](https://github.com/mozilla-frontend-infra/perf-goggles-demo/blob/master/src/containers/perfherder.jsx#L57)' for metrics-graphics.

If you have other suggestions please let me know.

If you don't have another that's fine (I don't want you to spend too much time in this). I will downcast the date object to a string and be done.

wlach commented:
@armenzg Ah I see, that makes sense. There is indeed no standard option to encode data objects in json, but surely you can do some small amount of pre-processing on the json to make it match the API?

armenzg commented:
I fixed and tests now run consistently.

armenzg committed:
Support suites that have the same option but different extraOptions

Fetching data for dromaeo_dom returns more than one pgo job since it
used to run in the past with different extraOptions (it now runs with e10s & stylo).

armenzg commented on src/utils/perfherder.js:
```diff
@@ -131,15 +138,14 @@ const prepareData = async (subtestsInfo) => {
-    perfherderUrl: perherderGraphUrl(signatureIds),
-    data: subtestsPerfData,
-  };
+  return data;
```
The refactor of this function is to help me when I produce data for `test/expected`.

wlach commented on src/utils/helper.js:
```diff
@@ -0,0 +1,19 @@
+// Based on https://stackoverflow.com/a/43478439/513672
+const arraysEquality = (_arr1, _arr2) => {
```
Could you just use lodash's isEqual here?

https://lodash.com/docs/4.17.10#isEqual

Supposedly it supports arrays

armenzg replied:
I'm not too familiar with `lodash`. Thanks for pointing it out to me.

wlach commented on src/App.jsx:
```diff
@@ -1,22 +1,16 @@
-      linux64: DEFAULT_BENCHMARKS,
-      'windows7-32': DEFAULT_BENCHMARKS,
-      'windows10-64': DEFAULT_BENCHMARKS,
+      linux64: ['ARES6', 'JetStream', 'dromaeo_dom', 'motionmark_htmlsuite', 'motionmark_animometer'],
```
Any reason you didn't just make `linux64: DEFAULT_BENCHMARKS + ['dromaeo_dom']`, instead of duplicating the definitions here?

armenzg replied:
Good catch!

wlach reviewed (commented):
looks ok to me in general, a few small comments
